src/common.jl

```diff
@@ -150,6 +150,10 @@ function SciMLBase.init(prob::LinearProblem, alg::SciMLLinearSolveAlgorithm,
         fill!(u0, false)
     end
 
+    # Guard against type mismatch for user-specified reltol/abstol
+    reltol = eltype(prob.A)(reltol)
+    abstol = eltype(prob.A)(abstol)
```

Comment:
```suggestion
    reltol = eltype(prob.b)(reltol)
    abstol = eltype(prob.b)(abstol)
```

A might be an operator.

Reply:
Nice catch, thanks!